File: csharp/sones-GraphMappingFramework/sones-GraphMappingFramework/ClientUtils/ServiceToken.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using sones.GraphMappingFramework.ServiceReference;

namespace sones.GraphMappingFramework.ClientUtils
{
    public class ServiceToken
    {
        public Nullable<long> CurrentTransactionToken { get; set; }
        
        //public RPCServiceContract ServiceReference { get; private set; }

        public GraphDS GraphDSService { get;  set; }
        
        public VertexTypeService VertexTypeService { get;  set; }
        
        public VertexInstanceService VertexInstanceService { get;  set; }
        
        public EdgeTypeService EdgeTypeService { get;  set; }
       
        public EdgeInstanceService EdgeInstanceService { get;  set; }

       
    }
}

File: java/sones-GraphMappingFramework/src/sones/GraphMapping/ServiceToken.java
package sones.GraphMapping;

import com.sones.EdgeInstanceService;
import com.sones.EdgeTypeService;
import com.sones.GraphDS;
import com.sones.RPCServiceContract;
import com.sones.VertexInstanceService;
import com.sones.VertexTypeService;

public class ServiceToken {
	private Long CurrentTransactionToken;
	public RPCServiceContract ServiceReference;
	public GraphDS GraphDSService;
	public VertexTypeService VertexTypeService;
	public VertexInstanceService VertexInstanceService;
	public EdgeTypeService EdgeTypeService;
	public EdgeInstanceService EdgeInstanceService;
	
	public Long getCurrentTransactionToken()
	{
		return CurrentTransactionToken;
	}
	
	protected void setCurrentTransactionToken(Long _CurrentTransactionToken)
	{
		CurrentTransactionToken = _CurrentTransactionToken;
	}
}
